ioctl/newcmd/alias/aliasset.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2020 IoTeX Foundation
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package alias
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+
+	"github.com/iotexproject/iotex-core/ioctl"
+	"github.com/iotexproject/iotex-core/ioctl/config"
+	"github.com/iotexproject/iotex-core/ioctl/output"
+	"github.com/iotexproject/iotex-core/ioctl/validator"
+)
+
+// Multi-language support
+var (
+	setCmdShorts = map[config.Language]string{
+		config.English: "Set alias for address",
+		config.Chinese: "设定地址的别名",
+	}
+	setCmdUses = map[config.Language]string{
+		config.English: "set ALIAS ADDRESS",
+		config.Chinese: "set 别名 地址",
+	}
+)
+
+// NewAliasSet represents the alias set command
+func NewAliasSetCmd(c ioctl.Client) *cobra.Command {
+
+	use, _ := c.SelectTranslation(setCmdUses)
+	short, _ := c.SelectTranslation(setCmdShorts)
+
+	ec := &cobra.Command{
+		Use:   use,
+		Short: short,
+		Args:  cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+			if err := validator.ValidateAlias(args[0]); err != nil {
+				return output.NewError(output.ValidationError, "invalid alias", err)
+			}
+			alias := args[0]
+			if err := validator.ValidateAddress(args[1]); err != nil {
+				return output.NewError(output.ValidationError, "invalid address", err)
+			}
+			addr := args[1]
+			aliases := GetAliasMap()
+			for aliases[addr] != "" {
+				delete(c.Config().Aliases, aliases[addr])
+				aliases = GetAliasMap()
+			}
+			c.Config().Aliases[alias] = addr
+			out, err := yaml.Marshal(c.Config())
+			if err != nil {
+				return output.NewError(output.SerializationError, "failed to marshal config", err)
+			}
+			if err := c.WriteFile(config.DefaultConfigFile, out, 0600); err != nil {
```

Comment:
do we support changing config file path? if so, here we shall write to user defined config file as well?

Reply:
yes, some ioctl commands can change config file. The old ioctl command doesn't seem to support user defined config dir. Do I need to add this feature?

Reply:
i think for newcmd, we should use user defined config if that's possible. for legacy ioctl cmd if it doesn't support this feature, it is ok.

Reply:
sure. I will add a new config cmd flag about custom dir. Maybe I can do it when migrate cmd/config.go cmd?